impl/src/main/java/org/ehcache/spi/serialization/DefaultSerializationProvider.java

```diff
@@ -63,23 +69,43 @@ public DefaultSerializationProvider(DefaultSerializationProviderConfiguration co
 
   @Override
   public <T> Serializer<T> createKeySerializer(Class<T> clazz, ClassLoader classLoader, ServiceConfiguration<?>... configs) throws UnsupportedTypeException {
+    Serializer<T> serializer;
     if (findSingletonAmongst(PersistenceSpaceIdentifier.class, (Object[]) configs) == null) {
-      return transientProvider.createKeySerializer(clazz, classLoader, configs);
+      serializer = transientProvider.createKeySerializer(clazz, classLoader, configs); 
     } else {
-      return persistentProvider.createKeySerializer(clazz, classLoader, configs);
+      serializer = persistentProvider.createKeySerializer(clazz, classLoader, configs);
     }
+    created.add(serializer);
+    return serializer;
   }
 
   @Override
   public <T> Serializer<T> createValueSerializer(Class<T> clazz, ClassLoader classLoader, ServiceConfiguration<?>... configs) throws UnsupportedTypeException {
+    Serializer<T> serializer;
     if (findSingletonAmongst(PersistenceSpaceIdentifier.class, (Object[]) configs) == null) {
-      return transientProvider.createValueSerializer(clazz, classLoader, configs);
+      serializer = transientProvider.createValueSerializer(clazz, classLoader, configs);
     } else {
-      return persistentProvider.createValueSerializer(clazz, classLoader, configs);
+      serializer = persistentProvider.createValueSerializer(clazz, classLoader, configs);
     }
+    created.add(serializer);
+    return serializer;
   }
 
   @Override
+  public void releaseSerializer(final Serializer<?> serializer) {
+    if (!created.remove(serializer)) {
+      throw new IllegalArgumentException("Given serializer: " + serializer.getClass().getName() + " is not managed by this provider");
+    }
+    if(serializer instanceof CompactPersistentJavaSerializer) {
```

Comment:
Having one more instances check for CompactJavaSerializer isn't enough? Reusing `Closable` has the side effect of ehcache closing any custom Serializer implementation that is closable. The next option is to have a new interface just for the sake of these 2 implementations. Is it really worth compared to having an additional instanceof check?